controllers/cluster/capabilities.go

```diff
@@ -0,0 +1,47 @@
+package cluster
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+)
+
+type Capabilities struct {
+	IsOnOpenshift, HasMachineAPI bool
+}
+
+func NewCapabilities(config *rest.Config) (Capabilities, error) {
+	caps := &Capabilities{}
+	err := caps.discover(config)
```

Comment:
Nit:
This pattern seems a bit of an over-complication (i.e discover filling in caps).
Wouldn't be simpler either:
```
func NewCapabilities(config *rest.Config) (Capabilities, error) {
	return discover(config)
}

func discover(config *rest.Config) (*Capabilities, error) { ..
```

Or even just drop discover and move the code into `NewCapabilities` ?

Reply:
yeah it makes sense, I will simplify it